Duplicati/Library/Utility/ReadLimitLengthStream.cs

```diff
@@ -50,8 +81,24 @@ public override int Read(byte[] buffer, int offset, int count)
         }
 
         public override long Seek(long offset, SeekOrigin origin)
-        {
-            return m_innerStream.Seek(offset, origin);
+        {
+            long innerPosition;
+            switch (origin)
+            {
+                case SeekOrigin.Begin:
+                    innerPosition = m_innerStream.Seek(m_start + offset, SeekOrigin.Begin);
+                    break;
+                case SeekOrigin.Current:
+                    innerPosition = m_innerStream.Seek(offset, SeekOrigin.Current);
+                    break;
+                case SeekOrigin.End:
+                    innerPosition = m_innerStream.Seek(m_start + m_length + offset, SeekOrigin.Begin);
+                    break;
+                default:
+                    throw new ArgumentException("Unknown SeekOrigin", nameof(origin));
+            }
+
+            return ClampInnerStreamPosition(innerPosition) - m_start;
```

Comment:
Since we have seeked `m_innerStream`, can we return `this.Position` to avoid duplicating the calculation?

Reply:
Indeed we can.